packages/common/src/hooks/useSortBy/useSortBy.ts

```diff
@@ -1,40 +1,51 @@
+import { KeyOf, ObjectWithStringKeys } from './../../types/utility';
 import { useCallback, useState } from 'react';
 
-export interface SortRule<T> {
-  key: keyof T;
+export interface SortRule<T extends ObjectWithStringKeys> {
+  key: KeyOf<T>;
   isDesc?: boolean;
 }
 
-export interface SortBy<T> extends SortRule<T> {
+export interface SortBy<T extends ObjectWithStringKeys> extends SortRule<T> {
   direction: 'asc' | 'desc';
 }
-export interface SortState<T> {
+export interface SortState<T extends ObjectWithStringKeys> {
   sortBy: SortBy<T>;
-  onChangeSortBy: (newSortRule: SortRule<T>) => void;
+  onChangeSortBy: (newSortRule: SortRule<T>) => SortBy<T>;
 }
 
-const getDirection = (isDesc: boolean) => (isDesc ? 'desc' : 'asc');
+const getDirection = (isDesc: boolean): 'asc' | 'desc' =>
+  isDesc ? 'desc' : 'asc';
 
-export const useSortBy = <T>({
+export const useSortBy = <T extends ObjectWithStringKeys>({
   key: initialSortKey,
   isDesc: initialIsDesc = false,
 }: SortRule<T>): SortState<T> => {
-  const [{ key, isDesc }, setSortBy] = useState({
+  const [sortBy, setSortBy] = useState<SortRule<T>>({
     key: initialSortKey,
     isDesc: initialIsDesc,
   });
 
   const onChangeSortBy = useCallback((newSortRule: SortRule<T>) => {
-    setSortBy(({ key: prevSortKey, isDesc: prevIsDesc }) => {
+    let newSortBy = sortBy;
+    setSortBy(({ key: prevSortKey, isDesc: prevIsDesc = false }) => {
       const { key: newSortKey, isDesc: maybeNewIsDesc } = newSortRule;
       const newIsDesc =
         prevSortKey === newSortKey ? !prevIsDesc : maybeNewIsDesc ?? false;
-      return { key: newSortKey, isDesc: newIsDesc };
+
+      newSortBy = {
+        key: newSortKey,
+        isDesc: newIsDesc,
+      };
+
+      return newSortBy;
     });
+
+    return { ...newSortBy, direction: getDirection(!!newSortBy?.isDesc) };
```

Comment:
This change is to return the calculated `SortBy` object when calling `onChangeSortBy`, which comes into play in `useSortedData`..